nixpkgs/toolchains/cc.nix

```diff
@@ -0,0 +1,284 @@
+        paths =
+          if stdenv.isDarwin then
+            [ (overrideCC stdenv darwinCC).cc darwin.binutils ]
+          else
+            [ stdenv.cc binutils ];
+        pathsToLink = [ "/bin" ];
+      }
+    else if isNull attribute_path then
+      nix_expr
+    else
+      lib.attrByPath (lib.splitString "." attribute_path) null nix_expr
+  ;
+in
+  runCommand "bazel-nixpkgs-cc-toolchain"
+    { executable = false;
+      # Pointless to do this on a remote machine.
+      preferLocalBuild = true;
+      allowSubstitutes = false;
+    }
+    ''
+      # This constructs the substitutions for
+      # `@bazel_tools//tools/cpp:BUILD.tpl` following the example of
+      # `@bazel_tools//tools/cpp:unix_cc_configure.bzl` as of Bazel v2.1.0 git
+      # revision 0f4c498a270f05b3896d57055b6489e824821eda.
+
+      # Determine toolchain tool paths.
+      #
+      # If a tool is not available then we use `bin/false` as a stand-in.
+      declare -A TOOLS=( [ar]=ar [cpp]=cpp [dwp]=dwp [gcc]=cc [gcov]=gcov [ld]=ld [nm]=nm [objcopy]=objcopy [objdump]=objdump [strip]=strip )
+      TOOL_NAMES=(''${!TOOLS[@]})
+      declare -A TOOL_PATHS=()
+      for tool_name in ''${!TOOLS[@]}; do
+        tool_path=${cc}/bin/''${TOOLS[$tool_name]}
+        if [[ -x $tool_path ]]; then
+          TOOL_PATHS[$tool_name]=$tool_path
+        else
+          TOOL_PATHS[$tool_name]=${coreutils}/bin/false
+        fi
+      done
+      cc=''${TOOL_PATHS[gcc]}
+
+      # Check whether a flag is supported by the compiler.
+      #
+      # The logic checks whether the flag causes an error message that contains
+      # the flag (or a pattern) verbatim. The assumption is that this will be a
+      # message of the kind `unknown argument: XYZ`. This logic is copied and
+      # adapted to bash from `@bazel_tools//tools/cpp:unix_cc_configure.bzl`.
+      is_compiler_option_supported() {
+        local option="$1"
+        local pattern="''${2-$1}"
+        { $cc "$option" -o /dev/null -c -x c++ - <<<"int main() {}" 2>&1 1>/dev/null || true; } \
+          | grep -qe "$pattern" && return 1 || return 0
+      }
+      is_linker_option_supported() {
+        local option="$1"
+        local pattern="''${2-$1}"
+        { $cc "$option" -o /dev/null -x c++ - <<<"int main() {}" 2>&1 1>/dev/null || true; } \
+          | grep -qe "$pattern" && return 1 || return 0
+      }
+      add_compiler_option_if_supported() {
+        if is_compiler_option_supported "$@"; then
+          echo "$1"
+        fi
+      }
+      add_linker_option_if_supported() {
+        if is_linker_option_supported "$@"; then
+          echo "$1"
+        fi
+      }
+
+      # Determine default include directories.
+      #
+      # This is copied and adapted to bash from
+      # `@bazel_tools//tools/cpp:unix_cc_configure.bzl`.
+      IFS=$'\n'
+      include_dirs_for() {
+        $cc -E -x "$1" - -v "''${@:2}" 2>&1 \
+          | sed '1,/^#include <...>/d;/^[^ ]/,$d;s/^ *//' \
```

Comment:
```suggestion
          | sed -e '1,/^#include <...>/d;/^[^ ]/,$d;s/^ *//' -e 's: (framework directory)::g' \
```

The CC toolchain did not work for me initially, after some debugging I found this in my `external/nixpkgs_config_cc/builtin_include_directory_paths`: 

```
/nix/store/0hxxabqbk48h756igafr4isd63i4ng5c-apple-framework-CoreFoundation/Library/Frameworks (framework directory)                                                                                                                                                                                                                                                                                                                                                                                       
/nix/store/bdkhqk4618bcicpd77icpr3lggdsxin3-apple-framework-Security/Library/Frameworks (framework directory)                                                                                                                                                                                                                                                                                                                                                                                             
/nix/store/d9dm4wgi5shhjzjjvgvgvvkjmcy79a0y-apple-framework-CoreServices/Library/Frameworks (framework directory)                                                                                                                                                                                                                                                                                                                                                                                         
/nix/store/h144jawqa92rqjhaahrsikq5j2dwkh5n-Libsystem-osx-10.12.6/include                                                                                                                                                                                                                                                                                                                                                                                                                                 
/nix/store/jgb1p0333mnbxibxhbqw8ql9apywbky1-apple-framework-Foundation/Library/Frameworks (framework directory)                                                                                                                                                                                                                                                                                                                                                                                           
/nix/store/nrsa1l0kh3vxindfkfa1d5fw62yvac4d-libc++-7.1.0/include/c++/v1                                                                                                                                                                                                                                                                                                                                                                                                                                   
/nix/store/sw7b2gcklaj3iy5ikv7ag163bpry3ivb-clang-wrapper-7.1.0/resource-root/include 
```

So it seems to be returned by cc:

```
/nix/store/pph3wv596llarhfvi1q6q2ija58dna5w-bazel-nixpkgs-cc/bin/cc -E -x c - -v "" 2>&1 | sed '1,/^#include <...>/d;/^[^ ]/,$d;s/^ *//'
/nix/store/0hxxabqbk48h756igafr4isd63i4ng5c-apple-framework-CoreFoundation/Library/Frameworks (framework directory)
/nix/store/d9dm4wgi5shhjzjjvgvgvvkjmcy79a0y-apple-framework-CoreServices/Library/Frameworks (framework directory)
/nix/store/bdkhqk4618bcicpd77icpr3lggdsxin3-apple-framework-Security/Library/Frameworks (framework directory)
/nix/store/jgb1p0333mnbxibxhbqw8ql9apywbky1-apple-framework-Foundation/Library/Frameworks (framework directory)
/nix/store/nrsa1l0kh3vxindfkfa1d5fw62yvac4d-libc++-7.1.0/include/c++/v1
/nix/store/sw7b2gcklaj3iy5ikv7ag163bpry3ivb-clang-wrapper-7.1.0/resource-root/include
/nix/store/h144jawqa92rqjhaahrsikq5j2dwkh5n-Libsystem-osx-10.12.6/include
```

Reply:
Good catch!